src/dfx/src/commands/canister/logs.rs

```diff
@@ -91,10 +116,62 @@ fn format_canister_logs(logs: FetchCanisterLogsResponse, opts: &LogsOpts) -> Vec
         .collect()
 }
 
+pub async fn exec(env: &dyn Environment, opts: LogsOpts, call_sender: &CallSender) -> DfxResult {
+    let callee_canister = opts.canister.as_str();
+    let canister_id_store = env.get_canister_id_store()?;
+
+    let canister_id = Principal::from_text(callee_canister)
+        .or_else(|_| canister_id_store.get(callee_canister))?;
+
+    fetch_root_key_if_needed(env).await?;
+
+    if opts.follow {
+        let interval = opts.interval.unwrap_or(2);
+        let mut last_idx = 0u64;
+
+        loop {
+            let logs = canister::get_canister_logs(env, canister_id, call_sender).await?;
+            let filter_opts = FilterOpts {
+                tail: None,
+                since: None,
+                since_time: None,
+                last_idx: Some(last_idx),
+            };
+            let filtered_logs = filter_canister_logs(&logs, filter_opts);
+
+            if !filtered_logs.is_empty() {
+                println!("{}", format_canister_logs(filtered_logs).join("\n"));
+                last_idx = filtered_logs.last().unwrap().idx;
+            }
+
+            tokio::select! {
+                _ = tokio::time::sleep(Duration::from_secs(interval)) => continue,
+                _ = tokio::signal::ctrl_c() => break,
+            }
+        }
+    } else {
+        let logs = canister::get_canister_logs(env, canister_id, call_sender).await?;
+
+        let filter_opts = FilterOpts {
+            tail: opts.tail,
+            since: opts.since,
+            since_time: opts.since_time,
+            last_idx: None,
+        };
+        let filtered_logs = filter_canister_logs(&logs, filter_opts);
+
+        if filtered_logs.is_empty() {
+            println!("No logs");
+        } else {
+            println!("{}", format_canister_logs(filtered_logs).join("\n"));
+        }
```

Comment:
I'd suggest printing nothing if there are no logs. Unless I'm missing something, this is the previous behavior.

```suggestion
        if !filtered_logs.is_empty() {
            println!("{}", format_canister_logs(filtered_logs).join("\n"));
        }
```